### PR TITLE
IAM Policy Creation

### PR DESCRIPTION
**Pull request by benJrohrer**
Implements creation of GooglePolicy classes from json, adds members and roles attributes

### CODE REVIEW

File: test/test_google_cloud_iam.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
+
+from cirrus.google_cloud.iam import (
+    GooglePolicyMember,
+    GooglePolicyRole,
+    GooglePolicyBinding,
+    GooglePolicy,
+)
+
+def test_create_google_policy():
+
```
Comment (Avantol13) [inline test/test_google_cloud_iam.py]: quick docstring here please. something like `"""Test that stuf works"""`

File: cirrus/google_cloud/iam.py [hunk 2]
```diff
@@ -20,9 +20,33 @@ def __init__(self, bindings, etag="", version=0):
             version (int): version for the policy
         """
         self.bindings = bindings
+        self.members = set()
```
Comment (Avantol13) [inline cirrus/google_cloud/iam.py]: Can we simplify to:
```
+        self.members = set()
+        self.roles = set()
+        for binding in bindings:
+            self.members.update(binding.members)
+            self.roles.add(binding.role)
```

File: cirrus/google_cloud/iam.py [hunk 3]
```diff
@@ -53,7 +77,34 @@ def __init__(self, role, members):
             members (List(GooglePolicyMember)): Member(s) who should have the given role
         """
         self.role = role
-        self.members = members
+        for m in members:
+            m.roles.add(role)
+        self.role.members.update(members)
+        self.members = set()
+        self.members.update(members)
```
Comment (Avantol13) [inline cirrus/google_cloud/iam.py]: if you want to save a line, you could just do `self.members = set(members)`

File: test/test_google_cloud_iam.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        "serviceAccount", "mem12@iam.gserviceaccount.com")
+    mem13 = GooglePolicyMember("user", "mem13@gmail.com")
+
+    role2 =  GooglePolicyRole("roles/read")
+    mem21 = GooglePolicyMember("user", "mem21@gmail.com")
+    mem22 = GooglePolicyMember(
+        "serviceAccount", "mem22@iam.gserviceaccount.com")
+    mem23 = GooglePolicyMember("user","mem23@gmail.com")
+
+    mems1 = [mem11, mem12, mem13, mem23]
```
Comment (Avantol13) [inline test/test_google_cloud_iam.py]: why is `mem23` in `mems1`? seems like by naming convention `mems1` group should only have `mem1x` members

File: test/test_google_cloud_iam.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    GooglePolicyMember,
+    GooglePolicyRole,
+    GooglePolicyBinding,
+    GooglePolicy,
+)
+
+def test_create_google_policy():
+
+    role1 = GooglePolicyRole("roles/admin")
+    mem11 = GooglePolicyMember("user", "mem11@gmail.com")
```
Comment (Avantol13) [inline test/test_google_cloud_iam.py]: could be a personal thing, but I find the use of numbers here require more brain power than I want to comprehend what's going on. I'd find `member_a, member_b, ...` and `group_x` and `group_y` or something more helpful. At a minimum, some comments about what you're setting up and testing would be helpful

File: test/test_google_cloud_iam.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+                            "serviceAccount:mem22@iam.gserviceaccount.com",
+                            "user:mem23@gmail.com"
+                        ]
+                }
+            ]
+    }
+
+    policy = GooglePolicy.from_json(json)
+
+    role1 = GooglePolicyRole("roles/admin")
```
Comment (Avantol13) [inline test/test_google_cloud_iam.py]: why create these from the Objects? Shouldn't the `policy` have everything created already? Don't you just need to assert that the `policy.bindings` and `policy.members` have what you expect?

File: test/test_google_cloud_iam.py (hunk 4 above)
Comment (benJrohrer) [reply test/test_google_cloud_iam.py]: I wanted to have one member be in both policy bindings. But the naming is confusing here, so i'll clean that up.

File: test/test_google_cloud_iam.py (hunk 5 above)
Comment (benJrohrer) [reply test/test_google_cloud_iam.py]: I added comments to make this more clear

**Pull request merged**